monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/commands/local_linux_curl.js

```diff
@@ -1,13 +1,13 @@
 import {OS_TYPES} from '../OsTypes';
 
 
-export default function generateLocalLinuxCurl(ip, osType) {
+export default function generateLocalLinuxCurl(ip, osType, username) {
     let bitText = osType === OS_TYPES.LINUX_32 ? '32' : '64';
-    return `curl https://${ip}:5000/api/monkey/download/monkey-linux-${bitText} -k
-    -o monkey-linux-${bitText};
-    chmod +x monkey-linux-${bitText};
-    ./monkey-linux-${bitText} m0nk3y -s ${ip}:5000\`;`;
+    let runAsUser = '';
+    if (username != '')
+      runAsUser = `runuser -u ${username} -- `;
```

Comment:
Runuser works only if you're admin, so I think you need to add `sudo` at the start

Reply:
Or even better, we should do `su - username -c "command"` because it can be ran without admin credentials